castnet/__init__.py

```diff
@@ -358,9 +360,33 @@ def convert_dtypes(self, value, attr_name, label="Injection"):
             return convert_datetime(value)
         return param_type(value)
 
-    def generic_post(self, request):
+    @staticmethod
+    def add_history(query, resource_id, method, requester=None, json_request=None):
+        params = {}
+        params["timeStamp"] = str(datetime.now().isoformat())
+        params["requester"] = requester
+        params["method"] = method
+        params["resourceId"] = resource_id
+        if method == "PATCH" or method == "POST":
+            query = (
+                query[:-13]
+                + """
+            \nWITH source
+            CREATE (n:historyRecord {timeStamp: $timeStamp, email: $requester, method: $method, resourceId: $resourceId, jsonRequest: $jsonRequest})-[:RESOURCE_ID]->(source)\n
+            """
+                + query[-13:]
+            )
+            params["jsonRequest"] = str(json_request)
+        else:
+            query += """
```

Comment:
This is for delete, right? And it's different because DELETE doesn't have the `return source`, correct? What if we added it in delete?